### PR TITLE
issue-2397 catchThrowableOfType enriched alternatives

### LINKED ISSUE #2397: `catchThrowableOfType` enriched alternatives
#### Summary
Similarly to the enriched alternatives of [`assertThatExceptionOfType`](https://assertj.github.io/doc/#assertj-core-exception-assertions-assertThatExceptionOfType), we could introduce enriched versions of `catchThrowableOfType`.

#### Example

```java
NullPointerException npe = catchNullPointerException( () -> ... );
IllegalArgumentException iae = catchIllegalArgumentException( () -> ... );
IllegalStateException ise = catchIllegalStateException( () -> ... );
IOException ioe = catchIOException( () -> ... );
```

### PR DESCRIPTION
**Pull request by Spacca**
#### Check List:
* Fixes #2397
* Unit tests : YES
* Javadoc with a code example (on API only) : YES
* PR meets the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by scordio** [comment]
Thanks for the PR @Spacca. I see you added `catch` versions for additional exception types, which could be totally fine from my perspective. We may want to align the corresponding `assertThat...Exception` methods, probably in a separate PR.

What do you think @joel-costigliola?

File: src/main/java/org/assertj/core/api/Assertions.java [hunk 1]
```diff
@@ -1380,6 +1380,222 @@ public static Throwable catchThrowable(ThrowingCallable shouldRaiseThrowable) {
...
+   * // succeeds as catchException returns null when the code does not throw any exceptions
+   * assertThat(catchException(() -&gt; {})).isNull();
+   *
+   * // fails as the thrown instance is not an Exception
+   * catchException(() -&gt; {throw new Throwable("boom!");});</code></pre>
+   *
+   * @param shouldRaiseException The lambda with the code that should raise the exception.
+   * @return The captured exception or <code>null</code> if none was raised by the callable.
+   * @see #catchThrowable(ThrowableAssert.ThrowingCallable)
+   * @since 3.21.0
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/Assertions.java:1404]: should be 3.22.0 (same of the other ones too)

**Comment by joel-costigliola** [comment]
We do want to align with `assertThat...Exception`, probably  in a different issue, in the new issue we could also add a test to verify that we have matching `assertThat...Exception` with `catch...Exception`.

**Comment by scordio** [comment]
Thanks for your first contribution, @Spacca!

**Pull request merged**